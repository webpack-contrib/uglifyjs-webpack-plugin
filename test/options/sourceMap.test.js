import webpack from '../helpers/compiler';
import UglifyJsPlugin from '../../src/index';

describe('Options', () => {
  describe('sourceMap', () => {
    test('{Boolean} - true', async () => {
      const config = {
        plugins: [
          new UglifyJsPlugin({
            sourceMap: true,
          }),
        ],
      };

      const stats = await webpack('entry.js', config);
      const { assets } = stats.compilation;

      const { source, map } = assets['main.bundle.js'].sourceAndMap();

      expect(source).toMatchSnapshot();
      expect(map).toMatchSnapshot();
    });
  });

  test('{Boolean} - true (parallel)', async () => {
    const config = {
      plugins: [
        new UglifyJsPlugin({
          parallel: true,
          sourceMap: true,
        }),
      ],
    };

    const stats = await webpack('entry.js', config);
    const { assets } = stats.compilation;

    const { source, map } = assets['main.bundle.js'].sourceAndMap();

    expect(source).toMatchSnapshot();
    expect(map).toMatchSnapshot();
  });
});
