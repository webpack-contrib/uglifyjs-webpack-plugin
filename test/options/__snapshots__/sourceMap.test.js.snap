// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options {Boolean} - true (parallel) 1`] = `
"(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(o,n){o.exports=function(){console.log(7)}}],[[0,1,0]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Options {Boolean} - true (parallel) 2`] = `
Object {
  "file": "x",
  "mappings": "",
  "names": Array [],
  "sources": Array [],
  "version": 3,
}
`;

exports[`Options sourceMap {Boolean} - true 1`] = `
"(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(o,n){o.exports=function(){console.log(7)}}],[[0,1,0]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Options sourceMap {Boolean} - true 2`] = `
Object {
  "file": "x",
  "mappings": "",
  "names": Array [],
  "sources": Array [],
  "version": 3,
}
`;
